src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -283,5 +283,24 @@ public override Uri GetUri(string name)
         {
             return new Uri(_directory.Uri, name);
         }
+
+        public override async Task<bool> IsUpdateRequiredAsync(Uri resourceUri, string id, string version)
+        {
+            //the destination blob
+            var blobName = resourceUri.Segments.Last();
+            var blobRelativePath = $"{id}/{version}/{blobName}";
+            var destination = _directory.GetBlockBlobReference(blobRelativePath);
+            if (await destination.ExistsAsync())
+            {
+                await destination.FetchAttributesAsync();
```

Comment:
Do we really need `FetchAttributesAsync` if we just called `ExistsAsync`? They should both be possible with a single HTTP request.

Reply:
Updated